internal/asm/f64/gemv_test.go

```diff
@@ -0,0 +1,616 @@
+				wantRevXY: []float64{168.4, 309.6, 275.2, 196.32, 316.2, 352.28, 309.36, 314.52, 292.84, 333.44, 295.16},
+			},
+		},
+
+		Trans: []DgemvSubcase{ // (4x4, 2x4, 1x4, 4x2, 2x2, 1x2, 4x1, 2x1, 1x1)
+			{alpha: 0, beta: 0,
+				want:      []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+				wantRevX:  []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+				wantRevY:  []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+				wantRevXY: []float64{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			},
+			{alpha: 0, beta: 1,
+				want:      []float64{2.5, 1.2, 0.8, 2.9, 3.4, 1.8, 4.6, 3.3, 3.8, 0.9, 1.1},
+				wantRevX:  []float64{2.5, 1.2, 0.8, 2.9, 3.4, 1.8, 4.6, 3.3, 3.8, 0.9, 1.1},
+				wantRevY:  []float64{2.5, 1.2, 0.8, 2.9, 3.4, 1.8, 4.6, 3.3, 3.8, 0.9, 1.1},
+				wantRevXY: []float64{2.5, 1.2, 0.8, 2.9, 3.4, 1.8, 4.6, 3.3, 3.8, 0.9, 1.1},
+			},
+			{alpha: 1, beta: 0,
+				want:      []float64{37.07, 55.58, 46.05, 47.34, 33.88, 54.19, 50.85, 39.31, 31.29, 55.31, 46.98},
+				wantRevX:  []float64{38.11, 63.38, 46.44, 40.04, 34.63, 59.27, 50.13, 35.45, 28.26, 51.64, 46.22},
+				wantRevY:  []float64{46.98, 55.31, 31.29, 39.31, 50.85, 54.19, 33.88, 47.34, 46.05, 55.58, 37.07},
+				wantRevXY: []float64{46.22, 51.64, 28.26, 35.45, 50.13, 59.27, 34.63, 40.04, 46.44, 63.38, 38.11},
+			},
+			{alpha: 8, beta: -6,
+				want:      []float64{281.56, 437.44, 363.6, 361.32, 250.64, 422.72, 379.2, 294.68, 227.52, 437.08, 369.24},
+				wantRevX:  []float64{289.88, 499.84, 366.72, 302.92, 256.64, 463.36, 373.44, 263.8, 203.28, 407.72, 363.16},
+				wantRevY:  []float64{360.84, 435.28, 245.52, 297.08, 386.4, 422.72, 243.44, 358.92, 345.6, 439.24, 289.96},
+				wantRevXY: []float64{354.76, 405.92, 221.28, 266.2, 380.64, 463.36, 249.44, 300.52, 348.72, 501.64, 298.28},
+			},
+		},
+	},
+}
+
+func TestGemv(t *testing.T) {
+	for _, test := range DgemvCases {
+		t.Run(fmt.Sprintf("(%vx%v)", test.m, test.n), func(tt *testing.T) {
+			for i, cas := range test.NoTrans {
+				tt.Run(fmt.Sprintf("NoTrans case %v", i), func(st *testing.T) {
+					// Test that it passes with row-major
+					dgemvcomp(st, test, false, cas, i)
+				})
+			}
+			for i, cas := range test.Trans {
+				tt.Run(fmt.Sprintf("Trans case %v", i), func(st *testing.T) {
+					// Test that it passes with row-major
+					dgemvcomp(st, test, true, cas, i)
+				})
+			}
+		})
+	}
+}
+
+func dgemvcomp(t *testing.T, test DgemvCase, trans bool, cas DgemvSubcase, i int) {
+	const (
+		xGdVal, yGdVal, aGdVal = 0.5, 1.5, 10
+		gdLn                   = 4
+	)
+	if trans {
+		test.x, test.y = test.y, test.x
+	}
+	prefix := fmt.Sprintf("Test (%vx%v) t:%v (a:%v,b:%v)", test.m, test.n, trans, cas.alpha, cas.beta)
+	xg, yg := guardVector(test.x, xGdVal, gdLn), guardVector(test.y, yGdVal, gdLn)
+	x, y := xg[gdLn:len(xg)-gdLn], yg[gdLn:len(yg)-gdLn]
+	ag := guardVector(test.A, aGdVal, gdLn)
+	a := ag[gdLn : len(ag)-gdLn]
+
+	lda := uintptr(test.n)
```

Comment:
I always test also with `lda > n`. I'd prefer to have it but I'm not sure how important it is.

Reply:
I didn't set up a copy/guard for the matrix, mainly because tests in `lapack` and `stat` cover the `lda > n` case.